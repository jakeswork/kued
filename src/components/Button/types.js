import { bool, node, shape } from 'prop-types';

export const propTypes = {
  children: node,
  secondary: bool,
  flat: bool,
  classes: shape({}).isRequired,
};

export const defaultProps = {
  children: null,
  secondary: false,
  flat: false,
};
