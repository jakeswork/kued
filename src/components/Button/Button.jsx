import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes, defaultProps } from './types';

const Button = ({
  classes,
  secondary,
  flat,
  children,
  link,
  ...props
}) => {
  let className = classes.button;

  if (secondary) className = classes.secondary;

  if (flat) className = classes.flat;

  if (link) return <Link role="button" {...props} className={className} to={link}>{ children }</Link>;

  return <button type="button" {...props} className={className}>{ children }</button>;
};

Button.propTypes = propTypes;

Button.defaultProps = defaultProps;

export default Button;
