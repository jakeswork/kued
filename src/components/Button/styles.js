const defaultStyles = theme => ({
  display: 'inline-block',
  textAlign: 'center',
  position: 'relative',
  padding: '.84375rem 1.5rem .65625rem',
  borderRadius: '.25rem',
  WebkitBoxShadow: '0 1px 5px 0 rgba(0,0,0,0.06)',
  boxShadow: '0 1px 5px 0 rgba(0,0,0,0.06)',
  fontSize: 16,
  fontWeight: 600,
  minWidth: 88,
  fontFamily: theme.fontFamily,
});

export default theme => ({
  button: {
    ...defaultStyles(theme),
    backgroundColor: theme.colorPrimary,
    border: '1px solid transparent',
    color: '#fff',
    textShadow: '0 1px 0 rgba(0,0,0,0.03)',
  },
  secondary: {
    ...defaultStyles(theme),
    backgroundColor: 'transparent',
    border: `1px solid ${theme.textSecondary}`,
    color: theme.colorPrimary,
  },
  flat: {
    ...defaultStyles(theme),
    color: theme.colorPrimary,
    border: 0,
    WebkitBoxShadow: 0,
    boxShadow: 0,
    textTransform: 'uppercase',
    background: 'transparent',
  },
});
