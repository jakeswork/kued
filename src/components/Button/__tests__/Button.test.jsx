import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

const defaultProps = {
  classes: {},
};

describe('The Button instance', () => {
  describe('when called with default properties', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<Button {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when the onClick prop is passed to the button', () => {
    describe('it should be called when the button is clicked', () => {
      const newProps = {
        ...defaultProps,
        onClick: jest.fn(),
      };
      const wrapper = shallow(<Button {...newProps} />);
      const button = wrapper.find('button');

      button.simulate('click');

      expect(newProps.onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('when style overrides are passed', () => {
    it('should add a style attribute to the element with the overrides', () => {
      const newProps = {
        ...defaultProps,
        style: {
          fontSize: 0,
        },
      };
      const wrapper = shallow(<Button {...newProps} />);
      const button = wrapper.find('button');

      expect(button.props().style).toEqual(newProps.style);
    });
  });
});
