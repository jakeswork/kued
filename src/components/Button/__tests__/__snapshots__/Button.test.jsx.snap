// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button instance when called with default properties should render without throwing errors 1`] = `
<button
  type="button"
/>
`;
